fastestimator/estimator.py

```diff
@@ -92,14 +94,28 @@ def fit(self, summary=None):
         return self._start()
 
     def _prepare_pipeline(self):
-        self.pipeline.global_batch_multiplier = get_num_devices()
-        self.pipeline.eval_shuffle = self.validation_steps is not None
-        self.pipeline.prepare()
-        self.do_eval = "eval" in self.pipeline.mode_list
+        if isinstance(self.pipeline, Scheduler):
+            do_eval = set()
+            for _, pipeline in self.pipeline.epoch_dict.items():
+                self._configure_single_pipeline(pipeline)
+                do_eval.add("eval" in pipeline.mode_list)
+            assert len(do_eval) == 1, "inconsistent validation option between pipelines"
+            self.do_eval = do_eval.pop()
+        else:
+            self._configure_single_pipeline(self.pipeline)
+            self.do_eval = "eval" in self.pipeline.mode_list
+            self.pipeline = Scheduler({0: self.pipeline})
```

Comment:
if you're going to wrap self.pipeline into a scheduler anyways, why not do this first and then avoid the if branching?

Reply:
fixed